### PR TITLE
Allow help command to accept extraneous parameters

### PR DESCRIPTION
**Pull request by Jun-How**
Instead of throwing an error when the user types in an incorrect parameter after help (e.g. `help 123` or `help addi`), `help` is simply executed to show the list of help commands. This fix will align with the UG about ignoring extraneous parameters for commands that do not take in parameters.

Will also change the UG portion to reflect only for single-word commands (specifically `exit` and `help`, not `delete all`).

ModTrekParser updated here as well to throw unknown command if empty input (spaces) are executed as command input.

### CODE REVIEW

File: src/main/java/seedu/modtrek/logic/parser/HelpCommandParser.java [hunk 1]
```diff
@@ -49,8 +45,7 @@ public HelpCommand parse(String args) throws ParseException {
         case "SORT":
             return new HelpCommand(SortCommand.MESSAGE_USAGE);
         default:
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
+            return new HelpCommand("");
```
Comment (cjyothika) [inline src/main/java/seedu/modtrek/logic/parser/HelpCommandParser.java:48]: Perhaps you can have another constructor for this case of empty message strings?

**Review by cjyothika — COMMENTED** [review]
Perhaps you can prevent bugs further by introducing a new HelpCommand constructor or define the empty string message in the HelpCommand class?

File: src/main/java/seedu/modtrek/logic/parser/HelpCommandParser.java (hunk 1 above)
Comment (Jun-How) [reply src/main/java/seedu/modtrek/logic/parser/HelpCommandParser.java:48]: What do you mean? What kind of bugs are you foreseeing?

Comment (cjyothika) [reply src/main/java/seedu/modtrek/logic/parser/HelpCommandParser.java:48]: I think avoiding manual input for the creation of new objects is good practice? (but as far as bugs, I think eg. someone might not want to the full help command message and have the empty string as his help command input).

Comment (cjyothika) [reply src/main/java/seedu/modtrek/logic/parser/HelpCommandParser.java:48]: It is a bit weird for someone reading the code to see the empty string input, as they might not immediately understand. @Jun-How

Comment (Jun-How) [reply src/main/java/seedu/modtrek/logic/parser/HelpCommandParser.java:48]: I see, what if I replace it with `HelpCommand.SHOWING_ALL_MESSAGE_USAGE` instead so `selectedMessage` will be updated to it? Then `selectedMessage` will simply be returned.

**Review by cjyothika — APPROVED** [review]
LGTM. Code looks cleaner now with less if branches.

**Pull request merged**